api/src/resources/dataset.rs

```diff
@@ -146,3 +156,25 @@ pub(crate) struct GetAvailableResponse {
 pub(crate) struct GetResponse {
     pub dataset: Dataset,
 }
+
+#[derive(Debug, Clone, Serialize)]
+pub struct UpdateDataset<'request> {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub source_ids: Option<&'request [SourceId]>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub title: Option<&'request str>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<&'request str>,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub(crate) struct UpdateRequest<'request> {
+    pub dataset: UpdateDataset<'request>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub(crate) struct UpdateResponse {
+    pub dataset: Dataset,
+}
```

Comment:
Are these structs there so that the `serde`-generated `json` matches the `api` schema?

Reply:
Yes, exactly. We could probably reduce the number of structs we have if we really wanted to (I think `UpdateResponse` and `CreateResponse` are actually the same), but for now I'm happy sticking to the 1:1 mapping until it gets painful for the sake of clarity.